pkg/collector/scheduler/scheduler.go

```diff
@@ -166,6 +169,14 @@ func (s *Scheduler) Stop(timeout ...time.Duration) error {
 	// Interrupt the main loop, proceeding to shut down all the queues
 	// `done` is buffered so we can proceed and wait for shutdown (or timeout)
 	s.done <- true
+
+	// Signal an exit to any remaining goroutine still trying to enqueue one-time checks,
+	// and wait for them to exit
+	close(s.cancelOneTime)
+	s.wgOneTime.Wait()
+	s.cancelOneTime = make(chan bool)
```

Comment:
Why re-create the channel and workgroup?

Reply:
that was done to support `scheduler.Reload`, but since I removed it we don't have to re-create these. Going to remove them 👍